agent/form_uploader.go

```diff
@@ -54,6 +55,21 @@ func (u *FormUploader) Upload(artifact *api.Artifact) error {
 		return err
 	}
 
+	if u.conf.DebugHTTP {
+		// If the request is a multi-part form, then it's probably a
+		// file upload, in which case we don't want to spewing out the
+		// file contents into the debug log (especially if it's been
+		// gzipped)
+		var requestDump []byte
+		if strings.Contains(request.Header.Get("Content-Type"), "multipart/form-data") {
+			requestDump, err = httputil.DumpRequestOut(request, false)
+		} else {
+			requestDump, err = httputil.DumpRequestOut(request, true)
+		}
+
+		u.logger.Debug("\nERR: %s\n%s", err, string(requestDump))
```

Comment:
Is this log output in the right order? Should we be showing the request dump first then the error? Or is this how the other places do it?

Reply:
It’s the same order as the other instance in this file:

https://github.com/buildkite/agent/blob/f99832974503f0d4dd4bff8145d86e2b0f00f2c4/agent/form_uploader.go#L88-L91